Sources/StreamChat/APIClient/Endpoints/Payloads/IdentifiableModel.swift

```diff
@@ -0,0 +1,61 @@
+//
+// Copyright © 2022 Stream.io Inc. All rights reserved.
+//
+
+import CoreData
+import Foundation
+
+typealias PreWarmedCache = [String: [String: NSManagedObjectID]]
+
+extension PreWarmedCache {
+    func model<T: NSManagedObject & IdentifiableModel>(for id: String, context: NSManagedObjectContext, type: T.Type) -> T? {
+        guard let objectId = self[T.className]?[id] else { return nil }
+        return try? context.existingObject(with: objectId) as? T
+    }
+}
+
+protocol IdentifiableModel {
+    static var className: String { get }
+    static var idKeyPath: String? { get }
+    static func id(for model: NSManagedObject) -> String?
+}
+
+private extension IdentifiableModel {
+    static var _className: String { String(describing: self) }
+}
+
+extension ChannelDTO: IdentifiableModel {
+    static let className: String = _className
+    static let idKeyPath: String? = "cid"
```

Comment:
```suggestion
    static let idKeyPath: String? = #keyPath(ChannelDTO.cid)
```
applies to all others as well